orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/job/JobUtils.java

```diff
@@ -57,14 +56,17 @@ public void cancelWait(StageExecution stage) {
       }
 
       String name = names.get(0);
-      Names parsedName = Names.parseName(name);
       Moniker moniker = (Moniker) stage.getContext().get("moniker");
       String appName;
 
       if (moniker != null) {
         appName = moniker.getApp();
       } else {
-        appName = (String) stage.getContext().getOrDefault("application", parsedName.getApp());
```

Comment:
Same here - I think `parsedName` should be the last fallback.

Reply:
I can validate that the application exists, the `stage.execution.application` won't be null so fallback to parseName won't be possible as long as I verify the application name.